src/Native/System.Native/pal_networking.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include "pal_types.h"
+
+/**
+ * These error values are different on every platform so make a
+ * platform-agnostic version that we convert to and send to managed
+ */
+enum GetAddrInfoErrorFlags
+{
+    PAL_EAI_AGAIN = 1,      // Temporary failure in name resolution.
+    PAL_EAI_BADFLAGS = 2,   // Invalid value for `ai_flags' field.
+    PAL_EAI_FAIL = 3,       // Non-recoverable failure in name resolution.
+    PAL_EAI_FAMILY = 4,     // 'ai_family' not supported.
+    PAL_EAI_NONAME = 5,     // NAME or SERVICE is unknown.
+};
+
+/**
+ * Flags to pass to GetNameInfo. These do not match
+ * from platform to platform and must be converted
+ */
+enum GetNameInfoFlags
+{
+    PAL_NI_NAMEREQD     = 0x1,
+    PAL_NI_NUMERICHOST  = 0x2,
+};
+
+/**
+ * Error codes from GetHostByName and GetHostByAddr
+ */
+enum GetHostErrorCodes
+{
+    PAL_HOST_NOT_FOUND = 1,
+    PAL_TRY_AGAIN = 2,
+    PAL_NO_RECOVERY = 3,
+    PAL_NO_DATA = 4,
+    PAL_NO_ADDRESS = PAL_NO_DATA,
+};
+
+struct PalIpAddress
```

Comment:
I started down that path and hit issues around determining length since IPv6 addresses can be variable size (::1 to full IPv6 with IPv4 mappings). Instead of dealing with that, this seemed like a cleaner and more easily understood solution, if it does have an extra indirection and one extra bit in the struct